ManPages/tsManPagesLibCLI/tsSysCommands.man
<open file '/usr/bin/logs/2015-04-20-at-06-25-49/pydoc.log', mode 'w' at 0x7fd63288>
Help on module tsSysCommands:

NAME
    tsSysCommands

FILE
    /cygdrive/d/WR/Technical_Preview-Draft/tsWxGTUI_PyVx/Python-2x/tsWxGTUI_Py2x/tsLibCLI/tsSysCommands.py

DESCRIPTION
    tsSysCommands.py - Class definition and methods for issuing
    shell commands to and receiving responses from the host
    operating system.

CLASSES
    __builtin__.object
        TsSysCommands
    
    class TsSysCommands(__builtin__.object)
     |  This class wraps the Python subprocess module which allows you to spawn
     |  new processes, connect to their input/output/error pipes, and obtain
     |  their return codes.
     |  
     |  Static methods defined here:
     |  
     |  tsCall(args, bufsize=0, executable=None, stdin=None, stdout=None, stderr=None, preexec_fn=None, close_fds=False, shell=False, cwd=None, env=None, universal_newlines=False, startupinfo=None, creationflags=0)
     |      Run command with arguments. Wait for command to complete,
     |      then return the returncode attribute.
     |      
     |      The arguments are the same as for the Popen constructor.
     |      
     |      Example: retcode = call(["ls", "-l"])
     |  
     |  tsCheckCall(args, bufsize=0, executable=None, stdin=None, stdout=None, stderr=None, preexec_fn=None, close_fds=False, shell=False, cwd=None, env=None, universal_newlines=False, startupinfo=None, creationflags=0)
     |      Run command with arguments. Wait for command to complete.
     |      If the exit code was zero then return, otherwise raise
     |      CalledProcessError. The CalledProcessError object will
     |      have the return code in the returncode attribute.
     |      
     |      The arguments are the same as for the Popen constructor.
     |      
     |      Example: check_call(["ls", "-l"])
     |  
     |  tsGetOutput(theCmd)
     |      Like getstatusoutput(), except the exit status is ignored and the
     |      return value is a string containing the command output.
     |  
     |  tsGetStatus(theFile)
     |      Return the output of "ls -ld theFile" as a string. This
     |      function uses the getoutput() function, and properly
     |      escapes backslashes and dollar signs in the argument.
     |  
     |  tsGetStatusOutput(cmd)
     |      Execute the string cmd in a shell with os.popen() and return a
     |      2-tuple (status, output). cmd is actually run as { cmd ; } 2>&1,
     |      so that the returned output will contain output or error messages.
     |      A trailing newline is stripped from the output. The exit status
     |      for the command can be interpreted according to the rules for
     |      the C function wait().
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors defined here:
     |  
     |  __dict__
     |      dictionary for instance variables (if defined)
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)

DATA
    CommandLineEnvWrapperEnable = True
    __authors__ = 'Richard S. Gordon'
    __copyright__ = 'Copyright (c) 2007-2013 Richard S. Gordon.\n\t\tAll r...
    __credits__ = '\n\n  Credits: \n\n\t  tsLibCLI Import & Application .....
    __date__ = '09/18/2013'
    __header__ = '\n\ntsSysCommands.py, v2.1.0 (build 09/18/2013)\n\n ...-...
    __license__ = 'GNU General Public License, Version 3, 29 June 2007'
    __line1__ = 'tsSysCommands.py, v2.1.0 (build 09/18/2013)'
    __line2__ = 'Author(s): Richard S. Gordon'
    __line3__ = 'Copyright (c) 2007-2013 Richard S. Gordon.\n\t\tAll right...
    __line4__ = 'GNU General Public License, Version 3, 29 June 2...7-2009...
    __title__ = 'tsSysCommands.py'
    __version__ = '2.1.0'
    mainTitleVersionDate = 'tsSysCommands.py, v2.1.0 (build 09/18/2013)'

VERSION
    2.1.0

DATE
    09/18/2013

CREDITS
    
    
      Credits: 
    
    	  tsLibCLI Import & Application Launch Features: 
    	  Copyright (c) 2007-2009 Frederick A. Kier.
    			All rights reserved.




Documents/NEWS.txt
#-----------------------------------------------------------
#"Time-stamp: <07/08/2015  4:48:41 AM rsg>"
#-----------------------------------------------------------

========================= NEWS.txt =========================

   +----+----+  TeamSTARS "tsWxGTUI_PyVx" Toolkit
   | ts | Wx |      with Python 2x & Python 3x based
   +----+----+         Command Line Interface (CLI)
   | G T U I |      and "Curses"-based "wxPython"-style, 
   +---------+         Graphical-Text User Interface (GUI)
 
   Get that cross-platform, pixel-mode "wxPython" feeling
   on character-mode 8-/16-color (xterm-family) & non-color
   (vt100-family) terminals and terminal emulators.

   You can find this and other plain-text files in the
   Toolkit subdirectory named:

       "./<Toolkit Recipient's Repository>/Documents".

       <Your Working Repository>
       (e.g. "tsWxGTUI_PyVx_Repository") 
         |
         |
         +-- ["Documents"]

=========================== News ===========================

The pre-alpha release of the TeamSTARS "tsWxGTUI_PyVx-0.0.0"
Toolkit is now available. It includes the following two
versions abd their associated variants (Python Distribution
Utilities ("Distutils") installable "Site-Packages" and non-
installable "Developer-Sandboxes" for enhancement and bug
fix development and experimentation without interference
with the already installed Site-Packages):

1. The TeamSTARS "tsWxGTUI_Py2x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   second generation Python 2.x programming language which
   is widely available.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. The final 2.x
   version 2.7 release came out in mid-2010, with a state-
   ment of extended support for this end-of-life release.
   The 2.x branch will see no new major releases after that.

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 2.0.0-2.7.9.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 2.6.4-2.7.9.

2. The TeamSTARS "tsWxGTUI_Py3x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   third generation Python 3.x programming language which
   is growing in availability.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. 3.x is under
   active development and has already seen over five years
   of stable releases, including version 3.3 in 2012 and
   3.4 in 2014. This means that all recent standard library
   improvements, for example, are only available by default
   in Python 3.x.

   Guido van Rossum (the original creator of the Python
   language) decided to clean up Python 2.x properly,
   with less regard for backwards compatibility than is
   the case for new releases in the 2.x range. The most
   drastic improvement is the better Unicode support
   (with all text strings being Unicode by default) as
   well as saner bytes/Unicode separation.

   Besides, several aspects of the core language (such as
   print and exec being statements, integers using floor
   division) have been adjusted to be easier for newcomers
   to learn and to be more consistent with the rest of the
   language, and old cruft has been removed (for example,
   all classes are now new-style, "range()" returns a memory
   efficient iterable, not a list as in 2.x).

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 3.0.0-3.4.2.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 3.1.5-3.4.2.

======================= End-Of-File ========================


SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-3x/tsWxGTUI_Py3x/tsLibCLI/tsSysCommandsPkg/src/tsSysCommands.py
#! /usr/bin/env python
#"Time-stamp: <02/24/2015  4:57:29 AM rsg>"
'''
tsSysCommands.py - Class definition and methods for issuing
shell commands to and receiving responses from the host
operating system.
'''
#################################################################
#
# File: tsSysCommands.py
#
# Purpose:
#
#     Class definition and methods for issuing shell commands
#     to and receiving responses from the host operating system.
#
# Limitations:
#
#     None
#
# Usage (examples):
#
#     python tsSysCommands
#
# Notes:
#
#     None
#
# Modifications:
#
#     2013/09/18 rsg Removed references to the now deprecated
#                    command module. Substituted the code from
#                    the Python-3.x version which had already
#                    substituted the appropriate subprocess
#                    module references.
#
# ToDo:
#
#     None
#
#################################################################

__title__     = 'tsSysCommands.py'

__version__   = '2.1.0'
__date__      = '09/18/2013'
__authors__   = 'Richard S. Gordon'
__copyright__ = 'Copyright (c) 2007-2013 ' + \
                '%s.\n\t\tAll rights reserved.' % __authors__
__license__   = 'GNU General Public License, ' + \
                'Version 3, 29 June 2007'
__credits__   = '\n\n  Credits: ' + \
                '\n\n\t  tsLibCLI Import & Application Launch Features: ' + \
                '\n\t  Copyright (c) 2007-2009 Frederick A. Kier.' + \
                '\n\t\t\tAll rights reserved.'
 
__line1__ = '%s, v%s (build %s)' % (__title__, __version__, __date__)
__line2__ = 'Author(s): %s' % __authors__
__line3__ = '%s' % __copyright__

if ((len(__credits__) == 0)):
    __line4__ = '%s' % __license__
else:
    __line4__ = '%s%s' % (__license__, __credits__)

__header__ = '\n\n%s\n\n  %s\n  %s\n  %s\n' % (__line1__,
                                               __line2__,
                                               __line3__,
                                               __line4__)

mainTitleVersionDate = __line1__

#--------------------------------------------------------------------------

from optparse import OptionParser
import os
import os.path
import subprocess
import sys

#--------------------------------------------------------------------------

try:

    import tsLibCLI

except ImportError as importCode:

    print('%s: ImportError (tsLibCLI); ' % __title__ + \
          'importCode=%s' % str(importCode))

#--------------------------------------------------------------------------

try:

    import tsExceptions as tse
    import tsLogger as Logger
    from tsReportUtilities import TsReportUtilities as tsrpu

    CommandLineEnvWrapperEnable = True

except ImportError as importCode:

    print('%s: ImportError (tsLibCLI); ' % __title__ + \
          'importCode=%s' % str(importCode))

#---------------------------------------------------------------------------

class TsSysCommands(object):
    '''
    This class wraps the Python subprocess module which allows you to spawn
    new processes, connect to their input/output/error pipes, and obtain
    their return codes.
    '''

    #----------------------------------------------------------------------

    @staticmethod
    def tsGetStatusOutput(cmd):
        '''
        Execute the string cmd in a shell with os.popen() and return a
        2-tuple (status, output). cmd is actually run as { cmd ; } 2>&1,
        so that the returned output will contain output or error messages.
        A trailing newline is stripped from the output. The exit status
        for the command can be interpreted according to the rules for
        the C function wait().
        '''
        try:
            return subprocess.getstatusoutput(cmd)
        except:
            return None
##            raise tse.UserInterfaceException(
##                tse.COMMAND_LINE_OPERATION_NOT_VALID,
##                'Command: "%s"' % cmd)

    #----------------------------------------------------------------------

    @staticmethod
    def tsGetOutput(theCmd):
        '''
        Like getstatusoutput(), except the exit status is ignored and the
        return value is a string containing the command output.
        '''
        try:
            return subprocess.getoutput(theCmd)
        except:
            return None
##            raise tse.UserInterfaceException(
##                tse.COMMAND_LINE_OPERATION_NOT_VALID,
##                'Command: "%s"' % theCmd)

    #----------------------------------------------------------------------

    @staticmethod
    def tsGetStatus(theFile):
        '''
        Return the output of "ls -ld theFile" as a string. This
        function uses the getoutput() function, and properly
        escapes backslashes and dollar signs in the argument.
        '''
        try:
            return subprocess.getstatus(theFile)
        except:
            return None
##            raise tse.UserInterfaceException(
##                tse.COMMAND_LINE_OPERATION_NOT_VALID,
##                'Command: "ls -ld %s"' % theFile)

    #----------------------------------------------------------------------

    @staticmethod
    def tsCall(args,
               bufsize=0,
               executable=None,
               stdin=None,
               stdout=None,
               stderr=None,
               preexec_fn=None,
               close_fds=False,
               shell=False,
               cwd=None,
               env=None,
               universal_newlines=False,
               startupinfo=None,
               creationflags=0):
        '''
        Run command with arguments. Wait for command to complete,
        then return the returncode attribute.

        The arguments are the same as for the Popen constructor.

        Example: retcode = call(["ls", "-l"])
        '''
 
        try:
            print('tsCall')
            return subprocess.call(args,
                                   bufsize=bufsize,
                                   executable=executable,
                                   stdin=stdin,
                                   stdout=stdout,
                                   stderr=stderr,
                                   preexec_fn=preexec_fn,
                                   close_fds=close_fds,
                                   shell=shell,
                                   cwd=cwd,
                                   env=env,
                                   universal_newlines=universal_newlines,
                                   startupinfo=startupinfo,
                                   creationflags=creationflags)
        except OSError as e:
            raise tse.ProgramException(tse.OS_ERROR,
                                       'Command: "%s"' % e)
        except ValueError as e:
            raise tse.UserInterfaceException(
                tse.COMMAND_LINE_OPERATION_NOT_VALID,
                'Command: "%s"' % e)

    #----------------------------------------------------------------------

    @staticmethod
    def tsCheckCall(args,
                    bufsize=0,
                    executable=None,
                    stdin=None,
                    stdout=None,
                    stderr=None,
                    preexec_fn=None,
                    close_fds=False,
                    shell=False,
                    cwd=None,
                    env=None,
                    universal_newlines=False,
                    startupinfo=None,
                    creationflags=0):
        '''
        Run command with arguments. Wait for command to complete.
        If the exit code was zero then return, otherwise raise
        CalledProcessError. The CalledProcessError object will
        have the return code in the returncode attribute.

        The arguments are the same as for the Popen constructor.

        Example: check_call(["ls", "-l"])
        '''
        try:
            print('tsCheckCall')
            subprocess.check_call(args,
                                  bufsize=bufsize,
                                  executable=executable,
                                  stdin=stdin,
                                  stdout=stdout,
                                  stderr=stderr,
                                  preexec_fn=preexec_fn,
                                  close_fds=close_fds,
                                  shell=shell,
                                  cwd=cwd,
                                  env=env,
                                  universal_newlines=universal_newlines,
                                  startupinfo=startupinfo,
                                  creationflags=creationflags)
        except TypeError as e:
            raise tse.UserInterfaceException(
                tse.COMMAND_LINE_OPERATION_NOT_VALID,
                'Command: "%s"' % e)
        except subprocess.CalledProcessError as e:
            raise tse.UserInterfaceException(
                tse.COMMAND_LINE_OPERATION_NOT_VALID,
                'Command: "%s"' % e)
        except KeyboardInterrupt:
            raise tse.UserInterfaceException(
                tse.ABORT_BY_OPERATOR,
                '')

#---------------------------------------------------------------------------
 
if __name__ == '__main__':

    print(__header__)
